tests/adapters/test_adapter.py

```diff
@@ -62,15 +110,33 @@ def test_not_implemented_methods(adapter, method, args):
                           ))
 def test_adapter_values(value, kwargs, result):
     a = FakeAdapter()
-    assert a.values(value, **kwargs) == result
+    with pytest.warns(FutureWarning):
+        assert a.values(value, **kwargs) == result
+
+
+def test_read_binary_values():
+    a = ProtocolAdapter([(None, "abcdefgh")])
+    assert list(a.read_binary_values()) == pytest.approx([1.6777999e+22, 4.371022e+24])
+
+
+def test_write_binary_values():
```

Comment:
It seems what you are testing here is `Adapter._format_binary_values` in the ieee header format? Might be worth a comment, so people can make sense of that weird test string. I know I can't :sweat_smile:

Reply:
Done.